jwst/stpipe/step.py

```diff
@@ -1296,3 +1302,31 @@ def _get_suffix(suffix, step=None, default_suffix=None):
     if suffix is None and step is not None:
         suffix = step.name.lower()
     return suffix
+
+
+def get_disable_crds_steppars(default=None):
+    """Return either the explicit default flag or retrieve from the environment
+
+    If a default is not specified, retrieve the value from the environmental variable
+    `STPIPE_DISABLE_CRDS_STEPPARS`.
+
+    Parameters
+    ----------
+    default: str, bool, or None
+        Flag to use. If None, the environmental is used.
+
+    Returns
+    -------
+    flag: bool
+        True to disable CRDS STEPPARS retrieval.
+    """
+    truths =  ('true', 'True', 't', 'yes', 'y')
+    if default:
+        if isinstance(default, bool):
+            return default
+        elif isinstance(default, str):
+            return default in truths
+        raise ValueError(f'default must be string or boolean: {default}')
+
+    flag = os.environ.get('STPIPE_DISABLE_CRDS_STEPPARS', '')
+    return flag in truths
```

Comment:
```suggestion
    if isinstance(default, bool):
        return default

    if isinstance(default, str):
        return default in truths
    
    if default is None:
        flag = os.environ.get('STPIPE_DISABLE_CRDS_STEPPARS', '')
        return flag in truths
    
    raise ValueError(f'If supplied, default must be string or boolean: {default}')
```

Breaking up the conditions here makes it much easier to read